### ISSUE TITLE
Re-enable http test server Close() calls.

### ISSUE THREAD

**Issue by lavalamp** [kubernetes/kubernetes#24546]
https://github.com/golang/go/issues/12262 is fixed in go1.6.

It should now be safe to search the code base for "Uncomment when fix #19254" and uncomment the .Close() calls.

**Comment by MHBauer** [comment]
How difficult do you estimate this to be? I want to get into the code and start helping.

Besides uncommenting the `defer`s that follow the comment, is there a removal of manual calls to `close()` that needs to be done? I see this pattern all in test files. Is doing the close necessary, or is it done so that the tests are strictly correct?

I see:
```
λ ag 'Uncomment when fix #19254' | wc -l
     211
```

**Comment by AdoHe** [comment]
@MHBauer this is necessary, we should always call the `close`, there is a lot of manually work, and if you have bandwidth, you can take this, we would really appreciate that. Thanks :)

**Comment by zhouhaibing089** [comment]
also search this one `TODO: Uncomment next two lines and remove third line when fix #19254`

**Comment by zhouhaibing089** [comment]
It could be fixed with this simple command line:

```
$ find . -name "*.go" -exec sed -i -e 'N;s/\/\/ TODO: Uncomment when fix.*\n[^/]*\/\/ //g' {} \;
```

will submit a PR.
